modules/citrus-http/src/main/java/com/consol/citrus/http/message/CookieParser.java

```diff
@@ -0,0 +1,127 @@
+    private static final String DOMAIN = "Domain";
+    private static final String MAX_AGE = "Max-Age";
+    private static final String VERSION = "Version";
+
+    /**
+     * Converts cookies from a HttpEntity into Cookie objects
+     * @param httpEntity The message to convert
+     * @return A array of converted cookies
+     */
+    Cookie[] convertCookies(HttpEntity<?> httpEntity) {
+        final List<Cookie> cookies = new LinkedList<>();
+
+        List<String> inboundCookies = httpEntity.getHeaders().get("Set-Cookie");
+        if (inboundCookies != null) {
+            for (String cookieString : inboundCookies) {
+                Cookie cookie = convertCookieString(cookieString);
+                cookies.add(cookie);
+            }
+        }
+
+        return cookies.toArray(new Cookie[0]);
+    }
+
+    /**
+     * Converts a cookie string from a http header value into a Cookie object
+     * @param cookieString The string to convert
+     * @return The Cookie representation of the given String
+     */
+    private Cookie convertCookieString(String cookieString) {
+        Cookie cookie = new Cookie(getCookieParam(NAME, cookieString), getCookieParam(VALUE, cookieString));
+
+        if (cookieString.contains(COMMENT)) {
+            cookie.setComment(getCookieParam(COMMENT, cookieString));
+        }
+
+        if (cookieString.contains(PATH)) {
+            cookie.setPath(getCookieParam(PATH, cookieString));
+        }
+
+        if (cookieString.contains(DOMAIN)) {
+            cookie.setDomain(getCookieParam(DOMAIN, cookieString));
+        }
+
+        if (cookieString.contains(MAX_AGE)) {
+            cookie.setMaxAge(Integer.valueOf(getCookieParam(MAX_AGE, cookieString)));
+        }
+
+        if (cookieString.contains(SECURE)) {
+            cookie.setSecure(Boolean.valueOf(getCookieParam(SECURE, cookieString)));
+        }
+
+        if (cookieString.contains(VERSION)) {
+            cookie.setVersion(Integer.valueOf(getCookieParam(VERSION, cookieString)));
+        }
+
+        return cookie;
+    }
+
+    /**
+     * Extract cookie param from cookie string as it was provided by "Set-Cookie" header.
+     * @param param The parameter to extract from the cookie string
+     * @param cookieString The cookie string from the cookie header to extract the parameter from
+     * @return The value of the requested parameter
+     */
+    private String getCookieParam(String param, String cookieString) {
+        if (param.equals(NAME)) {
+            return cookieString.substring(0, cookieString.indexOf('='));
+        }
+
+        if (param.equals(VALUE)) {
+            if (cookieString.contains(";")) {
+                return cookieString.substring(cookieString.indexOf('=') + 1, cookieString.indexOf(';'));
+            } else {
+                return cookieString.substring(cookieString.indexOf('=') + 1);
+            }
+        }
+
+        if(SECURE.equals(param) && cookieString.contains(SECURE)) {
+            return "true";
```

Comment:
A bit verbose, but could be replaced with `String.valueOf(true)`

Reply:
Yeah, well... :thinking: 
I'll do it, but I think the intended use case of `String.valueOf()` is more like if you have a variable parameter so that's not clear whether the boolean value will be `true` or `false` (e.g. a recursive set) but with String as return type. :smiley: